### PR TITLE
[Merged by Bors] - feat(Analysis.Normed.Ring.SeminormFromBounded): add SeminormFromBounded

### PR DESCRIPTION
**Pull request by mariainesdff**
We prove [BGR, Proposition 1.2.1/2] : given a nonzero additive group seminorm on a commutative ring `R` such that for some positive `c : ℝ` and every `x y : R`, the inequality `f (x * y) ≤ c * f x * f y)` is satisfied, we create a ring seminorm on `R`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by PatrickMassot — COMMENTED** [review]
I was writing a review but I started editing the file to check that some remark was correct and ended up rewriting a lot. So I pushed a commit that you should feel free to revert if you don’t like it (my original review now appears as outdated comments). 

The change is to replace all `(f_mul : ∃ c > 0, ∀ x y, f (x * y) ≤ c * f x * f y)` by `{c : ℝ} (f_mul : ∀ x y, f (x * y) ≤ c * f x * f y)`. In particular the positivity of `c` never play any role, although this requires a proof in two successive lemma (hence the new `seminormFromBounded_aux`).

File: Mathlib/Analysis/Normed/Ring/SeminormFromBounded.lean [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+Copyright (c) 2024 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández
+-/
+import Mathlib.Analysis.Normed.Ring.Seminorm
+import Mathlib.Analysis.SpecialFunctions.Pow.Complex
+
+/-!
+# seminormFromBounded
+In this file, we prove [BGR, Proposition 1.2.1/2] : given a nonzero additive group seminorm on a
```
[the remark this answers is not in the document]
Comment (mariainesdff) [reply Mathlib/Analysis/Normed/Ring/SeminormFromBounded.lean:11]: I added the link.

**Comment by mariainesdff** [comment]
Thank you very much for these changes, Patrick!

**Comment by PatrickMassot** [comment]
bors merge

**Pull request closed without merging**